.github/workflows/flake8.yml

```diff
@@ -0,0 +1,35 @@
+name: Run flake8
+
+on:
+  # Runs on pushes targeting the default branch
```

Comment:
I think this (and the next workflow) should always run; we can always explicitly skip it via https://docs.github.com/en/actions/managing-workflow-runs-and-deployments/managing-workflow-runs/skipping-workflow-runs